### PR TITLE
Allows you to walk on walls under openspace

### PR DESCRIPTION
**Pull request by ghost**
:cl: Qustinnus
 add: Walls are now solid and walkable from all directions, how amazing
/:cl:

No more falling through walls. 

image

They have a somewhat lighter drop shadow compared to open_space that doesn't have a closed turf below, this allows for you to distinguish walls, and holes with walls under them.

needs https://github.com/tgstation/tgstation/pull/50831 to be merged

### CODE REVIEW

**Comment by Ryll-Ryll** [comment]
When you think about is, aren't walls just really tall floors anyway?

**Comment by PKPenguin321** [comment]
that's a really confusing image tbh, it's not conveyed well enough

**Comment by Qustinnus** [comment]
do you think its too light or too dark?

**Comment by PKPenguin321** [comment]
too light, i thought the wall you were walking on was on the same z as the one you were on

**Comment by Qustinnus** [comment]
Alright, let me make an update and I'll post it okay?

**Comment by XDTM** [comment]
This is cool but i think it might cause confusion even with the best shadowing.

In my opinion it's probably better for the long term to make walls automatically spawn a special floor turf on the above z-level and make them intrinsically connected (i.e. destroying it should break the wall below).

**Comment by Qustinnus** [comment]
@XDTM Cool idea, I'll explore it after https://github.com/tgstation/tgstation/pull/50831 is merged, as it is required for that to work.

**Comment by Qustinnus** [comment]
@XDTM I thought about it some more and realized that'll be a huge PITA because of all the wall types we have

**Comment by Dennok** [comment]
Dont see very big PITA in 

> In my opinion it's probably better in the long term to make walls automatically spawn a special floor turf on the above z-level and make them intrinsically connected (i.e. destroying it should break the wall below).

idea/
I try realise it some later.

**Comment by Qustinnus** [comment]
It's a PITA if you want the floor above the wall to actually look like the wall below, else it's just plating which defeats the point of it being a walkable wall.

**Comment by XDTM** [comment]
You could have a generic type and simply make it copy the wall icon on Initialize. For any particular mechanical interactions (e.g. reinforced wall-floors being harder to break than wooden wall-floors) you could make it redirect to the connected wall.

It would still be kind of iffy when decontructing walls, since you'd be making holes in the floor above this way. Perhaps they could be an additional layer below regular floor plating to prevent that.

**Comment by Qustinnus** [comment]
you dont want the wall icon as people can't tell if its a floor or a wall

**Comment by XDTM** [comment]
true, my idea in that case would be using the wall icon with an extra overlay on top to mark it as a floor. Maybe some kind of square with a grid in the middle.

**Comment by Qustinnus** [comment]
awaiting dennok's idea before I fix this up.

**Comment by Dennok** [comment]
> You could have a generic type and simply make it copy the wall icon on Initialize. For any particular mechanical interactions (e.g. reinforced wall-floors being harder to break than wooden wall-floors) you could make it redirect to the connected wall.
> 
> It would still be kind of iffy when decontructing walls, since you'd be making holes in the floor above this way. Perhaps they could be an additional layer below regular floor plating to prevent that.

We no more need COPY icons. if we need actual icon of another object.
We actual have way to show icon if object in another place and this used in openspace visualisation mechanics. This mechanics allow interraction with shown object. i disable this because it block lattice/catewalk/floor building on openspace. 
But this easely can be enabled for some special objects like walls, glass blocks, another things that blocks Zfall.
And this change can be completely separated from this pr as it only visual.

**Comment by Dennok** [comment]
Done mechanical part of implementation but visuals is bad.

**Comment by Qustinnus** [comment]
can you show me so I get an idea of what u mean @Dennok

**Comment by Dennok** [comment]
now i disable blackdrop and make wall under openspace interractable ( change adjacent to work with different Z (i disable ` z != T0.z' check inhttps://github.com/tgstation/tgstation/blob/add74bee56cfde91602eb8ee28e3cae5959bc4f6/code/_onclick/adjacent.dm#L34 and now cant completely implement work only with z, z+1, z-1))
Now wall looks like normal, with no backdrop.
Maybe it can work with bachdrop with mouse_opacity = 0 but interractable must be only things that block Zmove (walls, glass windows, etc) or we lost ability to build floor/catwalk on openspace.
I think it can be bypassed by use of `look down` verb (currently not implemented).

P.S. I see somewhere PR that implements Look UP verb, but now i cant find it, someone see it?

**Comment by Qustinnus** [comment]
is a hotkey i think

**Comment by Dennok** [comment]
Ah thanks. But i say about PR, to see it implementation.

**Comment by Qustinnus** [comment]
https://github.com/tgstation/tgstation/pull/50782#event-3297034066

**Comment by Dennok** [comment]
Thanks!

**Comment by Dennok** [comment]
multiZ wall breacking 1

**Comment by Qustinnus** [comment]
this doesn't solve the issue, walls still look like normal walls.

**Comment by Dennok** [comment]
#51160 testable

**Comment by kriskog** [comment]
merge conflict

**Comment by Rohesie** [comment]
Is this being superseded by Dennok's PR?

**Comment by Qustinnus** [comment]
I can't tell I couldn't really entail what his PR contains but I don't think it will. I was waiting to see if he'd PR what he was talking about but I didn't see anything akin to what I had, so probably not.

**Comment by Rohesie** [comment]
@Dennok ?

**Comment by Dennok** [comment]
My PR add multiZ interraction and look down.
My PR independent from this PR.

File: config/maps.txt [hunk 1]
```diff
@@ -48,4 +47,5 @@ map runtimestation
 endmap
 
 map multiz_debug
+	default
```
Comment (nemvar) [inline config/maps.txt:50]: very nice.

File: code/game/turfs/open/openspace.dm [hunk 2]
```diff
@@ -45,21 +46,35 @@ GLOBAL_DATUM_INIT(openspace_backdrop_one_for_all, /atom/movable/openspace_backdr
...
 	. = ..()
-	var/turf/T = below()
-	if(!T)
+	var/turf/below = below()
+	if(!below)
 		vis_contents.len = 0
 		if(prune_on_fail)
 			ChangeTurf(/turf/open/floor/plating, flags = CHANGETURF_INHERIT_AIR)
 		return FALSE
+	if(can_zFall(user, 1, below)) //If we can fall, this means we use the dark backrop
```
Comment (nemvar) [inline code/game/turfs/open/openspace.dm:69]: user isn't defined anywhere.

File: code/_onclick/hud/plane_master.dm [hunk 3]
```diff
@@ -39,6 +39,15 @@
 /obj/screen/plane_master/proc/clear_filters()
 	filters = list()
 
+/obj/screen/plane_master/openspace/light
+	name = "light open space plane master"
+	plane = OPENSPACE_BACKDROP_PLANE_LIGHT
+
+/obj/screen/plane_master/openspace/light/backdrop(mob/mymob)
```
Comment (nemvar) [inline code/_onclick/hud/plane_master.dm:46]: Add the filters on Initialize, otherwise this won't work with cameras.

**Review by nemvar — COMMENTED** [review]
Did you accidentally remove the code that actually prevents you from falling?

**Comment by Cyberboss** [comment]
This really needs a different icon

**Comment by Dennok** [comment]
sigh

**Pull request closed without merging**